### PR TITLE
changed date to input string

### PR DESCRIPTION
**Pull request by LBHABEBIC**
# What?

# Why?

# Notes

# Next steps

### CODE REVIEW

File: steps/id-and-residency/carer.tsx [hunk 1]
```diff
@@ -89,20 +87,8 @@ const step: ProcessStepDefinition<ResidentDatabaseSchema, "carer"> = {
...
-                    startDate.month
-                      ? formatDate(
-                          new Date(startDate.year, startDate.month - 1),
-                          "MMMM yyyy"
-                        )
-                      : startDate.year
-                  }`
-                : null;
+            renderValue(name: string): React.ReactNode {
+              return name;
```
Comment (erbridge) [inline steps/id-and-residency/carer.tsx]:
```suggestion
            renderValue(dateString: string): React.ReactNode {
              return dateString;
```

File: steps/id-and-residency/carer.tsx [hunk 2]
```diff
@@ -259,9 +246,10 @@ const step: ProcessStepDefinition<ResidentDatabaseSchema, "carer"> = {
       ComponentWrapper.wrapDynamic(
         new DynamicComponent({
           key: "carer-live-in-start-date",
-          Component: DateInput,
+          Component: TextInput,
           props: {
             name: "carer-live-in-start-date",
+            label: "Record at least month and year if possible eg MM / YYYY.",
```
Comment (erbridge) [inline steps/id-and-residency/carer.tsx]: This is a hint, not a label. the label is the thing you've made a heading above.

File: storage/ResidentDatabaseSchema.ts [hunk 3]
```diff
@@ -57,7 +57,7 @@ type ResidentDatabaseSchema = NamedSchema<
         hasCarer: string;
         type: string;
         isLiveIn: string;
-        liveInStartDate: { month?: number; year?: number };
+        liveInStartDate: string;
```
Comment (erbridge) [inline storage/ResidentDatabaseSchema.ts:60]: This change should really have a data migration added for it, or you'll have issues with older processes.

File: __tests__/features/end-to-end.steps.ts [hunk 4]
```diff
@@ -374,7 +374,7 @@ const processData = {
         hasCarer: "yes",
         type: "registered",
         isLiveIn: "yes",
-        liveInStartDate: { month: 1, year: 2019 },
+        liveInStartDate: "",
```
Comment (erbridge) [inline __tests__/features/end-to-end.steps.ts]: Put a real test value in here.

File: __tests__/features/end-to-end.steps.ts [hunk 5]
```diff
@@ -666,21 +666,10 @@ defineFeature(loadFeature("./end-to-end.feature"), (test) => {
...
-      );
-      (
-        await browser!.waitForEnabledElement({
-          name: "carer-live-in-start-date-year",
-        })
-      ).sendKeys(
-        processData.residents[
-          presentTenantRef
-        ].carer.liveInStartDate.year.toString()
+        processData.residents[presentTenantRef].carer.liveInStartDate.toString()
```
Comment (erbridge) [inline __tests__/features/end-to-end.steps.ts]:
```suggestion
        processData.residents[presentTenantRef].carer.liveInStartDate
```

File: storage/ResidentDatabaseSchema.ts (hunk 3 above)
Comment (LBHABEBIC) [reply storage/ResidentDatabaseSchema.ts:60]: I'm not sure how to do this.

File: steps/id-and-residency/carer.tsx (hunk 2 above)
Comment (LBHABEBIC) [reply steps/id-and-residency/carer.tsx]: Ok...just the new design a heading and this as a sentence above the box. Hint doesn't exist on Text Input. what else could i use?

File: storage/ResidentDatabaseSchema.ts (hunk 3 above)
Comment (LBHABEBIC) [reply storage/ResidentDatabaseSchema.ts:60]: I'm not sure how to do this.

File: steps/id-and-residency/carer.tsx (hunk 2 above)
Comment (erbridge) [reply steps/id-and-residency/carer.tsx]: Ah. Ok, that's annoying. I forget if `TextInput` supports passing an id for a label, in which case, use the heading, add a paragraph and then this. The label is the heading, not this help text.

File: steps/id-and-residency/carer.tsx [hunk 6]
```diff
@@ -259,9 +246,10 @@ const step: ProcessStepDefinition<ResidentDatabaseSchema, "carer"> = {
       ComponentWrapper.wrapDynamic(
         new DynamicComponent({
           key: "carer-live-in-start-date",
-          Component: DateInput,
+          Component: TextInput,
```
Comment (erbridge) [inline steps/id-and-residency/carer.tsx:269]: Delete the `DateInput` component if nothing uses it after this change.

File: steps/id-and-residency/carer.tsx (hunk 2 above)
Comment (erbridge) [reply steps/id-and-residency/carer.tsx]: Ok, it doesn't. I think the work around for right now is to move the heading into this label and put the help text underneath as a paragraph. Not the best UX, but fixing this isn't a quick fix (and is a side effect of not having built these inputs in `lbh-frontend-react` properly).

File: storage/ResidentDatabaseSchema.ts (hunk 3 above)
Comment (erbridge) [reply storage/ResidentDatabaseSchema.ts:60]: Bump the database schema version, and add a migration to `storage/migrations/data` (and don't forget to set it up in the index file). Look at previous migrations for hints.

**Pull request merged**